datashader/glyphs.py

```diff
@@ -31,47 +31,27 @@ def validate(self, in_dshape):
     @staticmethod
     @ngjit
     def _compute_x_bounds(xs):
-        if len(xs) == 0:
-            raise ValueError('x coordinate array is empty.')
         minval = maxval = xs[0]
-        idx = 1
-        while idx < len(xs) and np.isnan(minval) and np.isnan(maxval):
-            x = xs[idx]
+        for x in xs:
             if not np.isnan(x):
-                minval = maxval = x
-            idx += 1
-        while idx < len(xs):
-            x = xs[idx]
-            if not np.isnan(x):
-                if x < minval:
+                if np.isnan(minval) or x < minval:
                     minval = x
-                elif x > maxval:
+                if np.isnan(maxval) or x > maxval:
```

Comment:
Yes, I think this does the right thing at the slight cost of checking nan every loop.